Fix case change for facility name

alexMet opened the pull request:
### Summary

Fixes case changing for facility name.

![simplescreenrecorder](https://user-images.githubusercontent.com/3750511/96054866-4c671880-0e8b-11eb-892f-d4a234ae4f1f.gif)


### References

Fixes #7493 

----

### Contributor Checklist


PR process:

- [x] PR has the correct target branch and milestone
- [ ] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

Testing:

- [x] Contributor has fully tested the PR manually
- [x] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

### Reviewer Checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

alexMet committed:
Fix case change for facility name

jonboiser reviewed (changes requested):
This looks good, but could you make one change at this line

https://github.com/learningequality/kolibri/pull/7561/files#diff-ee5f0b1e2b649ee99cd8efdf30af63a949c2961ff128b9f9d900375a607df5cbR14

by removing the `.trim` from `v-model.trim="name"`? This will prevent the textbox from trimming whitespace when users are editing the name (which can be jarring)

alexMet committed:
Remove trim from facility name model

alexMet commented:
@jonboiser I removed it from `v-model` and i added in the check inside of `nameIsInvalidText`. Because, otherwise you could submit the form with whitespace and an error would occur.

jonboiser commented:
> I removed it from `v-model` and i added in the check inside of `nameIsInvalidText`. Because, otherwise you could submit the form with whitespace and an error would occur.

Great call! I think this issue #7565 that was just reported is because we did not use `v-model.trim` but _also_ didn't do a later check of the trimmed input.
